frontend/src/components/pages/AddProduct.js

```diff
@@ -0,0 +1,232 @@
+import React, { useState } from "react";
+import ProductDataService from "../../client/product";
+import styled from "styled-components";
+
+const AddProduct = () => {
+    const initialProductState = {
+        title: "",
+        description: "",
+        price: "",
+        stock: "",
+        isActive: true,
+        categoryId: ""
+
+    };
+    const [product, setProduct] = useState(initialProductState);
+    const [submitted, setSubmitted] = useState(false);
+
+    const handleInputChange = event => {
+        const { name, value } = event.target;
+        setProduct({...product, [name]: value});
+    };
+
+
+    const saveProduct = () => {
+        var data = {
+            title: product.title,
+            description: product.description,
+            price: product.price,
+            stock: product.stock,
+            isActive: product.isActive,
+            categoryId: product.categoryId
+        };
+
+        ProductDataService.create(data)
+            .then(response => {
+                setProduct({
+                    title: response.data.title,
+                    description: response.data.description,
+                    price: response.data.price,
+                    stock: response.data.stock,
+                    isActive: response.data.isActive,
+                    categoryId: response.data.categoryId
+                });
+                setSubmitted(true);
+                console.log(response.data);
+            })
+            .catch(err => {
+                console.log(err);
+            });
+    };
+
+    const newProduct = () => {
+        setProduct(initialProductState);
+        setSubmitted(false);
+    };
+
+    return (
+    <Container>    
+    <Form>
+        <div className="submit-form">
+            <h2>Cadastrar Produtos</h2>
+            <div className="line"></div>
+            {submitted ? (
+                <div>
+                    <h4>Produto cadastrado com sucesso</h4>
```

Comment:
É legal ter uma sequencia nos headings, então depois do `h2`, o legal é não pular diretamente para o `h4`, daria para colocar `h3`.

```suggestion
            <h2>Cadastrar Produtos</h2>
            <div className="line"></div>
            {submitted ? (
                <div>
                    <h3>Produto cadastrado com sucesso</h3>
```

Da uma lida nesse artigo sobre os headings (nessa seção é falado sobre as ordens): https://medium.com/swlh/html-heading-tags-best-practices-78a85f63df47#a4fe

Reply:
Eu não sabia dessa questão de hierarquia!! Já corrigi, muito bom o artigo!